src/library/export/dlgtrackexport.h

```diff
@@ -0,0 +1,62 @@
+#ifndef DLGTRACKEXPORT_H
+#define DLGTRACKEXPORT_H
+
+#include <future>
+
+#include <QDialog>
+#include <QString>
+#include <QScopedPointer>
+#include <QTime>
+
+#include "configobject.h"
+#include "library/export/trackexport.h"
+#include "library/export/ui_dlgtrackexport.h"
+
+// A dialog for interacting with the track exporter in an interactive manner.
+// Handles errors and user interactions.
+class DlgTrackExport : public QDialog,
```

Comment:
I would spit the dlg class into gui and controller. Alternative you may may pass the dialog in the constructor of TrackExport.